File: problem_025.py
#!/usr/bin/env python

print("Project Euler - Problem 25:")
print("What is the index of the first term in the Fibonacci sequence to contain 1000 digits?\n")

a = 1
b = 0
c = 0
index = 1
debug = False

while len(str(a)) < 1000:
    c = a
    a += b
    b = c
    index += 1
    if debug == True: print index, a

print("%d" % index)

File: problem_025.lua
#!/usr/bin/env lua

print("Project Euler - Problem 25")
print("What is the index of the first term in the Fibonacci sequence to contain 1000 digits?\n")

number = 1
numTmp = 0
numPre = 0

count = 1
digits = 6

while (digits < 1000) do
   numTmp = number
   number = number + numPre
   numPre = numTmp

   while (number > 1000000) do
      number = number / 10
      numPre = numPre / 10
      digits = digits + 1
   end

   --print(number)
   count = count + 1
end

print(count)
